component/main.jsonnet

```diff
@@ -153,6 +153,48 @@ local consoleRoutePatch =
 
 local tls = import 'tls.libsonnet';
 
+// If we deploy cert-manager Certificates, we annotate namespace
+// openshift-config with the `kyvernoAnnotation` defined in `tls.libsonnet`
+// through a ResourceLocker patch. This triggers the the Kyverno policy to
+// copy the cert-manager TLS secrets into namespace openshift-config.
+//
+// We add the ResourceLocker patch to ArgoCD sync-wave 5, so it's guaranteed
+// to be applied in the cluster after the certificate has been issued and
+// before the custom openshift console route config is applied.
+//
+// NOTE: Due to the current implementation of the resource locker component
+// library this prevents other components from also providing ResourceLocker
+// patches for the `openshift-config` namespace.
+local openshiftConfigNsAnnotationPatch =
+  local needsPatch = hostname != null && std.length(tls.certs) > 0;
+  if needsPatch then
+    local target = kube.Namespace('openshift-config');
+    local patch = {
+      metadata: {
+        annotations: tls.kyvernoAnnotation,
+      },
+    };
+    [
+      if obj.kind == 'ResourceLocker' then
```

Comment:
I tend to find `std.map()` easier to read than this combination of `for` and `if`, but i guess that's pure preference...

Reply:
I could do `std.map()` and the if/else, but I'm not sure that would be clearer.

Reply:
Looked at the `std.map()` + if/else variant locally. I find that variation harder to read in this case, so I'll keep the list comprehension.